pages/docs/tutorials/native/apple-framework.md

```diff
@@ -127,109 +136,124 @@ Most of collections are mapped to collections from the other side:
++ (instancetype)new __attribute__((availability(swift, unavailable, message="use object initializers instead")));
+- (NSNumber * _Nullable)memberP:(int32_t)p __attribute__((swift_name("member(p:)")));
 @end;
+```
+</div>
+
+The code is full of Objective-C attributes, which are intended to help
+using the Framework from both Objective-C and Swift languages.
+`DemoClazz`, `DemoInterface` and `DemoObject` are created for `Clazz`, `Interface` and `Object` 
+respectively. The `Interface` is turned into `@protocol`, both class and object are represented as
+`@interface`.
+The `Demo` prefix comes from the `-output` parameter
+of the `kotlinc` compiler and the Framework name. 
+You may spot that the nullable return type `Long?` is turned into `NSNumber*` in Objective-C.
+
+### Global Declarations from Kotlin
+
+All global functions from Kotlin
+are turned into `Demo` class in Objective-C/Swift, where `Demo` is the Framework name and set by
+the `-output` parameter of `kotlinc`.
+
+<div class="sample" markdown="1" mode="obj-c" theme="idea" data-highlight-only="1" auto-indent="false">
+
+```obj-c
+NS_ASSUME_NONNULL_BEGIN
 
 __attribute__((objc_subclassing_restricted))
 @interface Demo : KotlinBase
-
-+ (void)forIntegersB:(int8_t)b s:(int16_t)s i:(int32_t)i l:(int64_t)l 
-     __attribute__((swift_name("forIntegers(b:s:i:l:)")));
-     
-+ (void)forFloatsF:(float)f d:(double)d 
-     __attribute__((swift_name("forFloats(f:d:)")));
-     
-+ (NSString *)stringsStr:(NSString *)str 
-     __attribute__((swift_name("strings(str:)")));
-     
-@property (class, readonly) NSString *globalString;
++ (void)forIntegersB:(int8_t)b s:(int16_t)s i:(int32_t)i l:(NSNumber * _Nullable)l __attribute__((swift_name("forIntegers(b:s:i:l:)")));
++ (void)forFloatsF:(float)f d:(NSNumber * _Nullable)d __attribute__((swift_name("forFloats(f:d:)")));
++ (NSString *)stringsStr:(NSString * _Nullable)str __attribute__((swift_name("strings(str:)")));
++ (NSString * _Nullable)acceptFunF:(NSString * _Nullable (^)(NSString *))f __attribute__((swift_name("acceptFun(f:)")));
 @end;
-
-NS_ASSUME_NONNULL_END
 ```
 </div>
 
-The name `DemoDemoClazz` and `DemoDemoObject` are created for `DemoClazz` and `DemoObject` 
-Kotlin classes. The `Demo` prefix comes from the `-output` parameter
-of the `kotlinc` compiler and the Framework name. All global functions from Kotlin
-are turned into `Demo` class in Objective-C/Swift, where `Demo` is the Framework name again.
-You may notice, that Kotlin `String` and Objective-C `NSString` are mapped transparently.
+You may notice, that Kotlin `String` and Objective-C `NSString*` are mapped transparently.
 Similarly, `Unit` type from Kotlin is mapped to `void`. We see primitive types
-are mapped directly.
+are mapped directly. Non-nullable primitive types are mapped transparently.
+Nullable primitive types are mapped into `NSNumber*`.
+You may see both higher order functions `acceptFunF` and `supplyFun` are included,
```

Comment:
I don't see the second one.

Reply:
returned that function back in the samples above, thanks for noticing that